content/realtime/authentication.textile

```diff
@@ -77,42 +77,110 @@ Here is a straightforward example of using Basic Authentication to connect to Ab
+
+```[jsall]
+const realtime = new Ably.Realtime({ authUrl: '/auth' });
 ```
 
 ```[ruby]
-  realtime = Ably::Realtime.new(auth_url: 'http://my.website/auth')
+realtime = Ably::Realtime.new(auth_url: '/auth')
 ```
 
 ```[java]
-  ClientOptions options = new ClientOptions();
-  options.authUrl = "http://my.website/auth";
-  AblyRealtime realtime = new AblyRealtime(options);
+ClientOptions options = new ClientOptions();
+options.authUrl = "/auth";
+AblyRealtime realtime = new AblyRealtime(options);
 ```
 
 ```[objc]
-  ARTClientOptions *options = [[ARTClientOptions alloc] init];
-  options.authUrl = [NSURL URLWithString:@"http://my.website/auth"];
-  ARTRealtime *realtime = [[ARTRealtime alloc] initWithOptions:options];
+ARTClientOptions *options = [[ARTClientOptions alloc] init];
+options.authUrl = [NSURL URLWithString:@"/auth"];
+ARTRealtime *realtime = [[ARTRealtime alloc] initWithOptions:options];
 ```
 
 ```[swift]
-  let options = ARTClientOptions()
-  options.authUrl = NSURL(string: "http://my.website/auth")
-  let realtime = ARTRealtime(options: options)
+let options = ARTClientOptions()
+options.authUrl = NSURL(string: "/auth")
+let realtime = ARTRealtime(options: options)
 ```
 
 ```[csharp]
-  ClientOptions options = new ClientOptions();
-  options.AuthUrl = new Uri("http://my.website/auth");
-  AblyRealtime realtime = new AblyRealtime(options);
+ClientOptions options = new ClientOptions();
+options.AuthUrl = new Uri("/auth");
+AblyRealtime realtime = new AblyRealtime(options);
+```
+
+h4(#auth-options). How to specify auth options
+
+There are several "auth options":/realtime/authentication#auth-options you can specify along with @authUrl@ and @authCallback@:
+
+* @authMethod@ - when @authUrl@ is called, the default "GET" method will be used, unless "POST" is specified.
+* @authHeaders@ - allows you to pass additional headers as required, depending on your use case.
+* @authParams@ - allows you to pass additional query parameters, depending on your use case.
+
+```[jsall]
+const ably = new Ably.Realtime({ authUrl: "/auth", authMethod: "POST", authParams: { p1: param1, b: param2}, authHeaders: {h1: header1, h2: header2} });
+```
+
+h3(#auth-callback). Token refresh using @authCallback@
+
+For non-browser clients, an authentication callback is the preferred method of implementing token refresh, although browser clients can also use @authCallback@ if the use case requires it. This relies on your application communicating securely with your own servers to obtain a token.
+
+The following code sample shows an example of using an @authCallback@ in a client:
+
+```[jsall](code-editor:realtime/auth-token-callback)
+const https = require("https");
+require("dotenv").config();
+
+const options = {
+  hostname: process.env.HOSTNAME,
+  port: 443, // SSL
+  path: "/auth",
+  method: "POST",
+  headers: {
+    "User-Agent": "Realtime Client v0.2",
```

Comment:
It's not clear to me what this user agent is supposed to do or where the v0.2 comes from. If there's a valid reason to keep this then I'd say we need a comment to explain.

Reply:
This was mainly added as Glitch requires it - but not necessary for this example. Replaced with stub.